### ISSUE TITLE
Thank you for sharing!

### ISSUE THREAD

**Issue by esutton** [davidpadbury/power-meter-reader#2]
I am amazed that a little over 100 lines of python can do this.

I am trying to figure out a couple of issues:
- The left-most dial 0 read as zero, expected 4
- Next dial, dial 1 seems like it should read 4 rather than 3 since dial to right is past zero.
    - Dial 4 also seems to be hair past 4
    - Glass meter covering hazy?
    - Photo not perfectly horizontal? ( tilted slightly to right)

**ToDo:**
1) Do need to add rules if very close to number, for example 3.9 and next right-dial is past 0, then round-up to 4.0 ?
2) Do need to clean glass on outside and inside?  The glass unscrews I think?

Analyze output images

![2020-feb-01](https://user-images.githubusercontent.com/6115047/109530252-1e6e2380-7a7c-11eb-8a94-5209ef31a436.JPG)
![hand-edge-0](https://user-images.githubusercontent.com/6115047/109530629-89b7f580-7a7c-11eb-9ad2-a968d6df904e.jpg)
![intersection-0](https://user-images.githubusercontent.com/6115047/109530637-8c1a4f80-7a7c-11eb-986f-334f888e31ce.jpg)
![hand-edge-1](https://user-images.githubusercontent.com/6115047/109530760-b10ec280-7a7c-11eb-83a1-46fe18b19319.jpg)
![intersection-1](https://user-images.githubusercontent.com/6115047/109530782-b835d080-7a7c-11eb-9f92-735afb864ec7.jpg)
![hand-edge-2](https://user-images.githubusercontent.com/6115047/109530852-d0a5eb00-7a7c-11eb-9bb6-96f1cbade2a9.jpg)
![intersection-2](https://user-images.githubusercontent.com/6115047/109530868-d6033580-7a7c-11eb-9089-4affe77875db.jpg)

**Comment by pastacolsugo** [comment]
Not the author nor an expert, but I've taken a look at the code and I don't think the haze on your glass is that relevant, the image gets blurred as one of the first steps in the processing.
I believe the inaccuracies you are experiencing are because the program is calibrated for his gauge, which has a simpler shape of the arm (needle?).
If you want to experiment I believe you could look at [line 30](https://github.com/davidpadbury/power-meter-reader/blob/51d2ce218a80b7d2f36a680166ebce18aaeee143/power-meter-reader.py#L30) and play around with the threshold, minLineLength and maxLineGap.
If you can I suggest you to try out the values with a old images and see if it gets better.
Good luck!

**Comment by esutton** [comment]
@pastacolsugo thank you for the suggestion!

Is the purpose of blurred to remove noise from image?  To remove small artifacts such as dirt or reflections?

The dial hand edge detection is hard to understand on the first left-most dial image.

Intersection is about 1 when expected 4.
![intersection-0](https://user-images.githubusercontent.com/6115047/110392700-fa3bb500-802e-11eb-9f6b-fcc9e7d9fd96.jpg)


Hand Edge shows a short vertical line at 1 when expected radial line at 4.
![hand-edge-0](https://user-images.githubusercontent.com/6115047/110392386-78e42280-802e-11eb-88cf-78b665cc275d.jpg)

Blurred
![blurred-0](https://user-images.githubusercontent.com/6115047/110392554-bc3e9100-802e-11eb-9868-50915a7a793a.jpg)

Edges
![edges-0](https://user-images.githubusercontent.com/6115047/110392572-c2cd0880-802e-11eb-9cfc-118641061cc2.jpg)

Dial
![dial-0](https://user-images.githubusercontent.com/6115047/110392665-eb550280-802e-11eb-845e-6baf59918141.jpg)

I will look at line 30 and [Hough Line Transform](https://opencv-python-tutroals.readthedocs.io/en/latest/py_tutorials/py_imgproc/py_houghlines/py_houghlines.html):

**Comment by pastacolsugo** [comment]
Yes usually you apply a blur to reduce noise. Maybe you can look into working with only one of the three RGB channels, you can try a binary filter or just increasing contrast.
If it's a simple operation you could try to clean out the haze, wouldn't hurt for sure.
